### DISCUSSION TITLE
NPC Aggression Timer - Auto-mode

### DISCUSSION THREAD

**Discussion by CreativeTechGuy** [runelite/runelite#17010] [Ideas]
It'd be really great if the NPC Aggression Timer could be a bit smarter about automatically turning on and off so you don't need to do it manually or whitelist a ton of NPCs.

I'm thinking that the "Auto mode" would do:

* If the player enters combat, start a timer for A seconds, after that timer is up, if the player is still in combat, then automatically enable the Aggression Timer.
* If the player leaves combat, and the auto-mode was previously enabled, start a timer for B seconds, if during this timer the player enters combat again, cancel the pending timer. If the timer completes and was not canceled, automatically disable the Aggression Timer.

A & B could be user-configurable, or set to some sensible default like A=60, B=180.

_I'm willing to contribute this feature if it is approved._

Edit: The current "Hide when out of combat" setting is similar in spirit, but has two issues, it only hides the lines, not the timer and doesn't disable the expiration notification, but also, it actives far too quickly so if you were actually AFK when you came back and noticed that you were out of combat, the lines would have disappeared when you needed them. - Not sure about backwards compatibility issues, but it might make sense to change the functionality of that existing toggle to this new more robust version.

Edit 2: To make this even more useful, it'd need to only auto-active when nearby mobs which are actually aggressive. It'd be pointless if it was active when the nearby enemies aren't aggressive to you. To handle this, it'd probably need to determine who initiates combat rather than hardcoding a list of aggressive NPCs. So the activation criteria should probably check if the player receives a hitspat from an enemy that the player isn't facing. Which might be a good proxy for aggression.

**Comment by ghost** [comment]
agree
